### PR TITLE
Expose `was_eligible` in the referral set stats

### LINKED ISSUE #9827: [Feature]: Show referral set eligibility in the API
# Feature Overview

**In order to** ensure we can identify active / eligibe referral sets
**We will** add a flag
**So that** Console doesnt have to do this itself

## Specs
- https://github.com/vegaprotocol/specs/blob/cosmicelevator/protocol/0083-RFPR-on_chain_referral_program.md

## Tasks
A checklist of the tasks that are needed to develop the feature and meet the acceptance criteria and feature test scenarios.
- [ ] Derive whether a referral set is eligible based on the staking balance of the referrer
- [ ] Add a flag to the `referralSets` indicating if the set is currentl eligible (as of the last epoch)

## Feature test scenarios
Detailed scenarios that can be executed as feature tests to verify that the feature has been implemented as expected.

GIVEN (setup/context) 
WHEN (action) 
THEN (assertion) For example...
See [here](https://github.com/vegaprotocol/vega/tree/develop/core/integration) for more format information and examples.

### Additional Details (optional)
Any additional information that provides context or gives information that will help us develop the feature.

### PR DESCRIPTION
**Pull request by ValentinTrinque**
Close #9827

### CODE REVIEW

File: protos/sources/vega/events/v1/events.proto [hunk 1]
```diff
@@ -919,6 +919,8 @@ message ReferralSetStatsUpdated {
   string rewards_multiplier = 6;
   // Rewards factor multiplier for the trades.
   string rewards_factor_multiplier = 7;
+  // States if the referral set was eligible to be part of the referral program.
```
Comment (candida-d) [inline protos/sources/vega/events/v1/events.proto:922]:
```suggestion
  // Indicates if the referral set was eligible to be part of the referral program.
```

File: datanode/gateway/graphql/schema.graphql [hunk 2]
```diff
@@ -6228,6 +6228,8 @@ type ReferralSetStats {
   atEpoch: Int!
   "Unique ID of the party."
   partyId: ID!
+  "States if the referral set was eligible to be part of the referral program."
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:6231]:
```suggestion
  "Indicates if the referral set was eligible to be part of the referral program."
```

File: core/events/referral_set.go [hunk 3]
```diff
@@ -18,10 +18,11 @@ package events
 import (
 	"context"
 
+	"golang.org/x/exp/slices"
+
```
Comment (EVODelavega) [inline core/events/referral_set.go:22]: RMBL

File: datanode/sqlstore/migrations/0046_additional_referral_set_stats.sql [hunk 4]
```diff
@@ -5,10 +5,16 @@ alter table referral_set_stats
 alter table referral_set_stats
     add column rewards_factor_multiplier text not null default '0';
 
+alter table referral_set_stats
```
Comment (EVODelavega) [inline datanode/sqlstore/migrations/0046_additional_referral_set_stats.sql:8]: is this migration on mainnet already, or not? (I suspect it isn't, but still thought I'd comment)

Comment (ValentinTrinque) [reply datanode/sqlstore/migrations/0046_additional_referral_set_stats.sql:8]: It's not.

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**